devito/operator/operator.py

```diff
@@ -1054,6 +1054,16 @@ def comm(self):
         """The MPI communicator the arguments are collective over."""
         return self.grid.comm if self.grid is not None else MPI.COMM_NULL
 
+    @property
+    def opkwargs(self):
+        temp_registry = {v: k for k, v in platform_registry.items()}
+        platform = temp_registry[self.platform]
+
+        temp_registry = {v: k for k, v in compiler_registry.items()}
+        compiler = temp_registry[self.compiler.__class__]
+
+        return {'platform': platform, 'compiler': compiler, 'language': self.language}
```

Comment:
Why not `self.compiler` and `self.platform`

`opkwargs` is a bit vague since it only returns the "language" part